### PR TITLE
feat: Ticks for plots need to be hourly

### LINKED ISSUE #185: Ticks for plots need to be hourly
## Overview

Currently the ticks are very precise, let's make this hourly so that it's easier for user to see. No more that 15 number of ticks!

### PR DESCRIPTION
**Pull request by madhavmk**
* Compute ticks and labels for x-axis ensuring that number of ticks stays between 5 and 15.
* Standard interval steps of 1s, 5s, 30s, 1min, 5min, 30min, 1hr, and 5hr are used whenever possible. Additionally, the intervals start out at the closest multiple of step.
* Datetime formats with 0.01s, 1s, and 1hr precision are used accordingly.
* Remove n_ticks argument from _compute_ticks_and_labels()
* Fixes #185

### CODE REVIEW

**Comment by madhavmk** [comment]
## Description:
The plot ticks are now made at standard intervals of 1s, 5s, 30s, 1min, 5min, 30min, 1hr, and 5hr.
X axis labelling is formatted with 0.01s, 1s, and 1hr precision as appropriate.
The number of x axis ticks will be in a range between 5 and 15 (excluding first and last ticks).
Residual and Residuals_enu_components plots are shown below:
### Hourly precision
Screenshot 2023-11-21 190502
Screenshot 2023-11-21 190455
### Second precision example 1
Screenshot 2023-11-21 182327
Screenshot 2023-11-21 182302
### Second precision example 2
Screenshot 2023-11-21 183117
Screenshot 2023-11-21 183110

### Millisecond precision is supported, but algorithm does not support such low precision (need to verify)

**Comment by madhavmk** [comment]
Hello @johnbdesanto ,
Could you please let me know your feedback based on the plot examples I have attached? I hope this satisfies the requirements.

**Comment by carlosgjs** [comment]
> * Compute ticks and labels for x-axis ensuring that number of ticks stays between 5 and 15.
> * Standard interval steps of 1s, 5s, 30s, 1min, 5min, 30min, 1hr, and 5hr are used whenever possible. Additionally, the intervals start out at the closest multiple of step.
> * Datetime formats with 0.01s, 1s, and 1hr precision are used accordingly.
> * Remove n_ticks argument from _compute_ticks_and_labels()
> * Issue 185: [Ticks for plots need to be hourly #185](https://github.com/uw-ssec/offshore-geodesy/issues/185)

Can you add to the description:

Fixes: #185 

This will automatically close the issue when the PR is merged

File: src/gnatss/ops/qc.py [hunk 1]
```diff
@@ -69,16 +52,77 @@ def _compute_ticks_and_labels(
...
     time_max = data[time_col].max()
-    step = (time_max - time_min) / n_ticks
-    ticks = np.arange(time_min, time_max + step, step)
-    labels = np.apply_along_axis(_sec_to_iso, 0, ticks)
+    time_delta = time_max - time_min
+
+    # step defines the interval in seconds between x-axis ticks
+    step = None
+
+    # Standard intervals of 1s, 5s, 30s, 1min, 5min, 30min, 1hr, 5hr
```
Comment (carlosgjs) [inline src/gnatss/ops/qc.py:60]: Looks like this is different from that issue #185 requests:

"This issue propose a change to have the tick steps by hourly if the difference b/w min and max is more than 1 hour, or be minutes if the difference is less than an hour with n_ticks no more than the specified n_ticks."
"No more that 15 number of ticks!"

ie, I interpret it as something like this:
```python
MAX_INTERVALS = 15
if time_delta <= SECS_IN_HOUR:
    num_intervals = max(MAX_INTERVALS, time_delta/SECS_IN_MINUTE)
    fmt =  "%Y-%m-%dT%H:%M"
else:
    num_intervals = max(MAX_INTERVALS, time_delta/SECS_IN_HOUR)
    fmt =  "%Y-%m-%dT%H:00"

step = ceil(time_delta/num_intervals)
```

**Review by carlosgjs — CHANGES_REQUESTED** [review]
One clarification comment and looks like we are missing a unit test for `_compute_ticks_and_labels`

**Comment by johnbdesanto** [comment]
The plots look fine to me. Thanks for the update.

File: src/gnatss/ops/qc.py (hunk 1 above)
Comment (madhavmk) [reply src/gnatss/ops/qc.py:60]: Hello @carlosgjs . The main reason I did not add ticks as you mentioned above is I wanted to add tick labels at "good to read" intervals steps like 1s, 30s, 1min, 1hr etc.

**Comment by johnbdesanto** [comment]
I will add one more thing. Looking at the standard tick intervals being implemented, it would be better to have 6 hr intervals instead of 5 hr intervals since that would lead to 4 ticks per day.

**Comment by madhavmk** [comment]
> I will add one more thing. Looking at the standard tick intervals being implemented, it would be better to have 6 hr intervals instead of 5 hr intervals since that would lead to 4 ticks per day.

Good suggestion!
This is what the updated chart with 6-hour intervals looks like
Screenshot 2023-11-22 131910

**Comment by madhavmk** [comment]
Thank you for your code suggestions @carlosgjs 😁 
I have incorporated your PR comments and added unittests. Please approve.

**Review by johnbdesanto — APPROVED** [review]

**Review by lsetiawan — APPROVED** [review]
Thanks @madhavmk, @carlosgjs, @johnbdesanto for working on this. Looking through this all looks good on the code side and the plot results.

File: src/gnatss/ops/qc.py (hunk 1 above)
Comment (lsetiawan) [reply src/gnatss/ops/qc.py]: Thanks for the great interpretation @carlosgjs. Looks like this is now implemented, but with even finer resolution of seconds, when finer precision is required.

**Pull request merged**